homeassistant/components/device_tracker/ping.py

```diff
@@ -0,0 +1,116 @@
+"""
+Support for detect network host with ping.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.ping/
+
+device_tracker:
+  - platform: ping
+    count: 2
+    hosts:
+      - 192.168.2.10
+      - 192.168.2.25
+"""
+
+import logging
+import subprocess
+import re
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, DEFAULT_SCAN_INTERVAL)
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant import util
+from homeassistant import const
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = []
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PING_COUNT = 'count'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(const.CONF_HOSTS): cv.ensure_list,
```

Comment:
Instead of passing in hosts, you probably want to use the Python network module so people can pass in subnets and it can ping all of them in that range.

https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv4Network.hosts